data/blog/session-replay-sdk-bundle-size-optimizations.md

```diff
@@ -0,0 +1,208 @@
+
+- [@sentry/webpack-plugin](https://www.npmjs.com/package/@sentry/webpack-plugin)
+- [@sentry/vite-plugin](https://www.npmjs.com/package/@sentry/vite-plugin)
+- [@sentry/rollup-plugin](https://www.npmjs.com/package/@sentry/rollup-plugin)
+- [@sentry/esbuild-plugin](https://www.npmjs.com/package/@sentry/esbuild-plugin)
+
+You can just update to its latest version, and add this configuration to the plugin:
+
+```js
+sentryPlugin({
+  bundleSizeOptimizations: {
+    excludeDebugStatements: true,
+    excludeReplayIframe: true,
+    excludeReplayShadowDom: true,
+  },
+})
+```
+
+This will safe you about 5 KB gzipped of bundle size!
+
+### How to implement build-time tree shaking flags
+
+We already had some build-time flags for tree shaking implemented in the JavaScript SDK itself (`__SENTRY_DEBUG__` and `__SENTRY_TRACING__`). We followed the same structure for rrweb:
+
+```js
+// General tree shaking flag example
+if (typeof __SENTRY_DEBUG__ === 'undefined' || __SENTRY_DEBUG__) {
+  console.log('log a debug message!')
+}
+```
+
+By default, this code will result in `log a debug message!` being logged.
+However, if you replace the `__SENTRY_DEBUG__` constant at build time with `false`, this will result in the following code:
+
+```js
+if (typeof false === 'undefined' || false) {
+  console.log('log a debug message!')
+}
+```
+
+Which bundlers will optimize to the following:
+
+```js
+if (false) {
+  console.log('log a debug message!')
+}
+```
+
+And in turn, since the code inside of `if (false)` will definitely never be called, will be completely tree shaken away.
+
+For rrweb, we used the same approach to allow to remove certain recording managers:
+
+1. In order to avoid touching all the parts of the code that may use a manager, we added new dummy managers following the same interface but doing nothing:
+
+```ts
+interface ShadowDomManagerInterface {
+  init(): void
+  addShadowRoot(shadowRoot: ShadowRoot, doc: Document): void
+  observeAttachShadow(iframeElement: HTMLIFrameElement): void
+  reset(): void
+}
+
+class ShadowDomManagerNoop implements ShadowDomManagerInterface {
+  public init() {}
+  public addShadowRoot() {}
+  public observeAttachShadow() {}
+  public reset() {}
+}
+```
+
+2. Now, in the place where the ShadowDomManager is usually initialized, we can do the following:
+
+```js
+const shadowDomManager =
+  typeof __RRWEB_EXCLUDE_SHADOW_DOM__ === 'boolean' && __RRWEB_EXCLUDE_SHADOW_DOM__
+    ? new ShadowDomManagerNoop()
+    : new ShadowDomManager()
+```
+
+This means that by default, the regular `ShadowDomManager` is used. However, if you replace `__RRWEB_EXCLUDE_SHADOW_DOM__` at build time with `true`, the `ShadowDomManagerNoop` will be used, and the `ShadowDomManager` will thus be tree shaken away.
+
+## Removed canvas recording support by default
+
+Since we currently do not support replaying captured canvas elements, and because the canvas capturing code makes up a considerable amount of the rrweb codebase,
```

Comment:
I also added the link above, we could keep it just here perhaps.

```suggestion
Since we currently do [not support replaying captured canvas elements](https://github.com/getsentry/sentry-javascript/issues/6519), and because the canvas capturing code makes up a considerable amount of the rrweb codebase,
```